hybridse/src/codegen/udf_ir_builder_test.cc

```diff
@@ -115,6 +115,56 @@ TEST_F(UdfIRBuilderTest, dayofweek_date_udf_test) {
     Date date(2020, 05, 22);
     CheckUdf<int32_t, Date>("dayofweek", 6, date);
 }
+TEST_F(UdfIRBuilderTest, dayofyear_date_udf_test) {
+    {
+        Date date(2020, 05, 22);
+        CheckUdf<int32_t, Date>("dayofyear", 143, date);
+    }
+    {
+        Date date(2021, 01, 01);
+        CheckUdf<int32_t, Date>("dayofyear", 1, date);
+    }
+    {
+        Date date(2020, 12, 31);
+        CheckUdf<int32_t, Date>("dayofyear", 366, date);
+    }
+    {
+        Date date(2021, 12, 31);
+        CheckUdf<int32_t, Date>("dayofyear", 365, date);
+    }
+    {
+        Date date(2021, 13, 31);
+        CheckUdf<int32_t, Date>("dayofyear", 0, date);
```

Comment:
It would be better if you return NULL when using an invalid date string. Returning 0 is also acceptable. Please declare it in functions' documentation and add related demo SQL.

Reply:
Sounds good.

Reply:
Sure, we will work it out in https://github.com/4paradigm/OpenMLDB/issues/883.
Thanks again for your help.